### PR TITLE
download_manager: try alternate content-length header (bug 1843393)

### PR DESCRIPTION
**Pull request by zzzeid**
- try alternate content-length header to determine size
- fail gracefully if total size can not be determined
- update tests to support this new alternate header
- make content-length header name case more consistent

### CODE REVIEW

**Review by cgsheeh — DISMISSED** [review]

File: mozregression/download_manager.py [hunk 1]
```diff
@@ -163,8 +163,18 @@ def _download(self, url, dest, finished_callback, chunk_size, session):
...
-                self._update_progress(bytes_so_far, total_size)
+                # GCP storage does not always return a Content-length header, check alternates.
+                content_length_headers = [
+                    "Content-length",
+                    "x-goog-stored-content-length",
+                ]
+                total_size = 0
+                for header in content_length_headers:
+                    if header in response.headers:
+                        total_size = int(response.headers[header].strip())
```
[the remark this answers is not in the document]
Comment (cgsheeh) [reply mozregression/download_manager.py]: My approach wouldn't check the second header if the first one was found.

[the remark this answers is not in the document]
Comment (zzzeid) [reply mozregression/download_manager.py]:
```python
int(response.headers.get("Content-length", response.headers.get("x-goog-stored-content-length", 0)))
```

becomes:

```python
int(response.headers.get("Content-length", alternate_value)))
```

Where `alternate_value` is `response.headers.get("x-goog-stored-content-length", 0))`. That has to be evaluated before the final value is determined.

[the remark this answers is not in the document]
Comment (zzzeid) [reply mozregression/download_manager.py]: Simple way to illustrate order of calls in Python (inner calls get evaluated before outer calls):

```python
>>> test = mock.MagicMock()
>>> test.get("key1", test.get("key2"))
<MagicMock name='mock.get()' id='4358537952'>
>>> test.mock_calls
[call.get('key2'),
 call.get('key1', <MagicMock name='mock.get()' id='4358537952'>)]
```

File: tests/unit/test_download_manager.py [hunk 2]
```diff
@@ -26,7 +26,13 @@ def iter_content(chunk_size=4):
             rest = rest[chunk_size:]
             yield chunk
 
-    response.headers = {"Content-length": str(len(data))}
+    if alternate_content_length is False:
+        response.headers = {"Content-length": str(len(data))}
```
Comment (zzzeid) [inline tests/unit/test_download_manager.py:30]:
```suggestion
        response.headers = {"content-length": str(len(data))}
```

**Review by cgsheeh — DISMISSED** [review]

**Review by cgsheeh — APPROVED** [review]

**Pull request merged**